### PR TITLE
fix(agent, browsing): too long tool description for openai

### PR DESCRIPTION
**Pull request by xingyaoww**
**End-user friendly description of the problem this fixes or functionality that this introduces**

- [ ] Include this change in the Release Notes. If checked, you must provide an **end-user friendly** description for your change below

---
**Give a summary of what the PR does, explaining any non-trivial design decisions**

Fix for this issue due to OpenAI 1024 character tool description restriction ([link](https://community.openai.com/t/tool-calling-api-upgrade-1024-char-limit-is-limiting/951951/5)):

image

This PR also adds a retry for RemoteRuntime to get the initial pod status.

---
**Link of any specific issues this addresses**

---

To run this PR locally, use the following command:
```
docker run -it --rm   -p 3000:3000   -v /var/run/docker.sock:/var/run/docker.sock   --add-host host.docker.internal:host-gateway   -e SANDBOX_RUNTIME_CONTAINER_IMAGE=docker.all-hands.dev/all-hands-ai/runtime:67ba9c6-nikolaik   --name openhands-app-67ba9c6   docker.all-hands.dev/all-hands-ai/openhands:67ba9c6
```

### CODE REVIEW

**Comment by xingyaoww** [comment]
I ran on a subset of SWE-Bench Lite & it doesn't seem to impact our quality too much. I'd say this is probably safe to go.

# resolved: 42 / 97. (43.30%)

**Review by tofarr — APPROVED** [review]

File: openhands/runtime/impl/remote/remote_runtime.py [hunk 1]
```diff
@@ -277,6 +278,24 @@ def _wait_until_alive(self):
...
+                'GET',
+                f'{self.config.sandbox.remote_runtime_api_url}/runtime/{self.runtime_id}',
+            )
+            runtime_data = runtime_info_response.json()
+            assert 'runtime_id' in runtime_data
+            assert runtime_data['runtime_id'] == self.runtime_id
+            assert 'pod_status' in runtime_data
+            pod_status = runtime_data['pod_status']
+            n_attempts += 1
+
```
Comment (rbren) [inline openhands/runtime/impl/remote/remote_runtime.py:298]: Let's not do our own retry logic. Instead, you can just `raise RuntimeNotReadyException` below to trigger tenacity

Maybe we should move this into its own PR too

Comment (xingyaoww) [reply openhands/runtime/impl/remote/remote_runtime.py]: Reverted!

Comment (xingyaoww) [reply openhands/runtime/impl/remote/remote_runtime.py]: done - and move these stuff to https://github.com/All-Hands-AI/OpenHands/pull/4825
